sdk/trace/src/main/java/io/opentelemetry/sdk/trace/MonotonicClock.java

```diff
@@ -19,7 +19,7 @@
  * it is not recommended to use only one instance for a very long period of time.
  */
 @Immutable
-public final class MonotonicClock implements Clock {
```

Comment:
Just an FYI - We may want to use this clock for the upcoming "Timer" instrument in Metrics.

Reply:
Yeah I think we might move it to a public place again in some form. I want to give it more though at the time though, at least for now the biggest problem this PR is solving is that spans must have this clock (a rename to `TraceClock` might be warranted) and that it's not a `Clock` as per the SDK interface definition.

Reply:
`MonotonicClock` exists to be anchored to a trace - just want to confirm you need `MonotonicClock` for the `Timer` and not just `Clock.nanoTime()`? (it is a bit weird that `Clock` exposes a monotonic time method in addition to wall time, it's effectively both a normal and monotonic clock, just not anchored to a trace). The only purpose of `MonotonicClock` is to provide "more accurate" wall timestamps, through its specialization of `now()`, I wouldn't expect it to be needed to capture a duration.